### PR TITLE
Move PLP actions and parsing to the merlins connector

### PR DESCRIPTION
**Pull request by jnurse**
This PR moves the PLP actions and parsers into the merlin's connector

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1254
 **Linked PRs**: n/a

## Changes
- Added fetcher for PLP container
- Moved PLP parsers into the merlin's connector
- Moved product fetch action to the merlin's connector
- Deleted actions file for the products section of the store (no longer needed) 

## How to test-drive this PR
- Make sure you're using the merlins-connector in main.jsx
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- view any product list page and confirm the page still works as expected

### CODE REVIEW

File: web/app/containers/app/actions.js [hunk 1]
```diff
@@ -145,9 +142,6 @@ export const fetchPage = (url, pageComponent, routeName, fetchUrl) => {
                 } else if (pageComponent === ProductDetails) {
                     dispatch(productDetailsActions.process(receivedAction))
                     // dispatch(productsActions.processProductDetails(receivedAction))
-                } else if (pageComponent === ProductList) {
```
Comment (jeremywiebe) [inline web/app/containers/app/actions.js]: 😺

File: web/app/integration-manager/merlins-connector/categories/parser.test.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 /* eslint-env jquery, jest */
 import {jquerifyHtmlFile} from 'progressive-web-sdk/dist/test-utils'
-import productListParser from './product-list'
+import categoryProductsParser from './parser'
 
 describe('the product list parser', () => {
-    const $content = jquerifyHtmlFile('app/store/categories/parsers/product-list.test.html')
-    const parsedContent = productListParser($, $content)
+    const $content = jquerifyHtmlFile('app/integration-manager/merlins-connector/categories/parser.test.html')
```
Comment (jeremywiebe) [inline web/app/integration-manager/merlins-connector/categories/parser.test.js]: Using a long path makes this brittle.  Is it safe to use `${__dirname}/parser.test.html` here?

File: web/app/integration-manager/merlins-connector/commands.js [hunk 3]
```diff
@@ -2,11 +2,12 @@ import {jqueryResponse} from 'progressive-web-sdk/dist/jquery-response'
 import {makeRequest, makeFormEncodedRequest, makeJsonEncodedRequest} from 'progressive-web-sdk/dist/utils/fetch-utils'
 import {urlToPathKey} from '../../utils/utils'
 
-import {productDetailsParser} from '../../store/products/parser'
 import {pdpAddToCartFormParser} from './parsers'
 import {checkoutShippingParser, parseCheckoutData} from './checkout/parsers'
 import homeParser from './home/parser'
 import {parseNavigation} from './navigation/parser'
+import categoryProductsParser from './categories/parser'
+import {productListParser, productDetailsParser} from './products/parser'
```
Comment (jeremywiebe) [inline web/app/integration-manager/merlins-connector/commands.js]: This is a bit inconsistent with the previous line where one uses a default export and one uses named exports. Both filenames are called `parser.js` ... so maybe this one should be `parsers.js` to signal it contains more than one?  Or maybe we just always use named exports for parsers and always call them `parsers.js`?

**Comment by DanielBrookRoberge** [comment]
:+1: from me modulo the merge conflicts.

**Comment by jnurse** [comment]
@DanielBrookRoberge yeah I'm in the middle of re-working a few things related to the merge conflicts, it might be a good idea to take another look at it when I'm done.

**Pull request merged**